pkg/controller/discovery/config/config.go

```diff
@@ -0,0 +1,69 @@
+package config
+
+import (
+	"flag"
+	"os"
+	"path/filepath"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/purser/pkg/controller/utils"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Context holds Kubernetes cluster contexts info
+type Context struct {
+	Kubeclient *kubernetes.Clientset
+	Config     *rest.Config
+	Namespace  string
+	Service    string
+}
+
+// Kubernetes configuration
+var (
+	KUBECONFIG = filepath.Join(os.Getenv("HOME"), ".kube", "config")
+	Ctx        *Context
+)
+
+func init() {
+	kubeconfig := flag.String("kubeconfig", KUBECONFIG, "Path to the kubeconfig file")
+	ns := flag.String("namespace", "default", "Kubernetes namespace for the application")
+	loglevel := flag.String("loglevel", utils.INFOLEVEL, "loglevel")
+	svc := flag.String("service", "", "Kubernetes application service")
+	flag.Parse()
+
+	utils.InitializeLogger(loglevel)
+	log.SetLevel(utils.LOGLEVEL)
+
+	log.Infof("Using Kubernetes configuration: `%v`", *kubeconfig)
+	log.Infof("Running application dynamics against namespace: `%v`", *ns)
+	log.Infof("Getting client, config of the cluster...")
+
+	config, err := getConfig(*kubeconfig)
+	if err != nil {
+		log.Errorf("failed to load client config: %v", err)
+	}
+
+	// build the Kubernetes client
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Errorf("failed to create kubernetes client: %v", err)
+	}
+
+	Ctx = &Context{
+		Kubeclient: client,
```

Comment:
client, config can be retrieved using this already existing function: https://github.com/vmware/purser/blob/master/pkg/client/clientset.go#L33